src/screens/OrderChat/index.tsx

```diff
@@ -0,0 +1,194 @@
+/* eslint-disable react/style-prop-object */
+import React, {
+  useEffect, useState, useCallback, useRef,
+} from 'react';
+import {
+  View,
+  ActivityIndicator,
+} from 'react-native';
+import { FlatList } from 'react-native-gesture-handler';
+import { Feather } from '@expo/vector-icons';
+import { KeyboardAccessoryView } from 'react-native-keyboard-accessory';
+import { useAuth } from '../../hooks/useAuth';
+import { MessageItem } from './components/MessageItem';
+import { getChatInstance, getChatMessages, sendChatMessage } from '../../services/chat';
+import { isIphoneX } from '../../utils/helper';
+import { useKeyboard } from '../../hooks/useKeyboard';
+import * as S from './styles';
+
+interface OrderChatProps {
+  route: {
+    params: {
+      orderId: number
+    }
+  }
+}
+
+const OrderChat = ({ route: { params: { orderId } } }: OrderChatProps): JSX.Element => {
+  const { user } = useAuth();
+  const [loading, setLoading] = useState<boolean>(false);
+  const [isVisibleKeyboard] = useKeyboard();
+
+  const [chatInstance, setChatInstance] = useState(null);
+  const [messages, setMessages] = useState([]);
+  const [currentOffset, setCurrentoffset] = useState(10);
+
+  const [inputValue, setInputValue] = useState('');
+  const [refresh, setRefresh] = useState(true);
+  const timer = useRef(null);
+
+  const loadMessages = useCallback(
+    async () => {
+      setLoading(true);
+
+      if (!chatInstance) {
+        await getChatInstance({
+          object_type: 'technology-order',
+          object_id: orderId,
+          target_user: user.id,
+        }).then((chat) => {
+          if (chat) setChatInstance(chat);
+        });
+      }
+
+      if (!chatInstance) {
+        return;
+      }
+
+      let m;
+
+      await getChatMessages(chatInstance.id, { offset: 0 }).then((data) => {
+        m = merge(messages, data, 'id');
+        m = m.sort((a: any, b: any) => b.id - a.id);
```

Comment:
This lines are not looking so good. What does the merge function do? I think a reduce function would solve a lot of things here 🤔

Reply:
I did some refactoring with more descriptive names and simplified logic. Thanks!